samples/openthread/coprocessor/README.rst

```diff
@@ -72,6 +72,29 @@ You can also use your own application, provided that it supports the Spinel comm
     |thread_hwfc_enabled|
     In addition, the Co-processor sample reconfigures the baud rate to 1000000 bit/s by default.
 
+Diagnostic module
+=================
+
+By default, the Co-processor sample comes with the :kconfig:option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` :ref:`feature set <thread_ug_feature_sets>` enabled, which allows you to use ``diag`` commands.
+Use these commands to manually check hardware-related functionalities without running a Thread network.
+For example, to ensure radio communication is working when adding a new functionality or during the manufacturing process.
+
+.. note::
+    If you disable the :kconfig:option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` feature set, you can enable the diagnostic module with the :kconfig:option:`CONFIG_OPENTHREAD_DIAG` Kconfig option.
+
+Additionally for the ``nrf52840dongle_nrf52840`` build target, diagnostic GPIO commands have been enabled by ``openthread_config`` node in :file:`boards/nrf52840dongle_nrf52840.overlay`.
+This node has been configured for P0.19 pin, which is connected to reset on nRF52840 Dongle. By setting it in output mode with low state reset is pulled to GND, which results in device rebooting without skipping the bootloader.
+
```

Comment:
I'd add some info about why this is useful (we want to enter bootloader from cli and other possible ways to reset from cli fail to do so)

Reply:
added info in cli sample section